WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift

```diff
@@ -229,10 +233,53 @@ private extension PaymentCaptureOrchestrator {
                                  currency: order.currency,
                                  receiptDescription: receiptDescription(orderNumber: order.number),
                                  statementDescription: statementDescriptor,
+                                 receiptEmail: provideReceiptEmailIfNecessary(from: order),
                                  paymentMethodTypes: paymentMethodTypes,
                                  metadata: metadata)
     }
 
+    private func synchronizePlugins(from siteID: Int64) async throws {
+        return try await withCheckedThrowingContinuation({
+                (continuation: CheckedContinuation<(), Error>) in
+            let systemPluginsAction = SystemStatusAction.synchronizeSystemPlugins(siteID: siteID) { result in
+                if case let .failure(error) = result {
+                    DDLogError("[PaymentCaptureOrchestrator] Error syncing system plugins: \(error)")
+                    continuation.resume(throwing: error)
+                } else {
+                    continuation.resume(returning: ())
+                }
+            }
+            ServiceLocator.stores.dispatch(systemPluginsAction)
+            })
+    }
+
+    /// We do not need to set the receipt email if WCPay is installed and active
+    /// and its version is higher or equal than 4.0.0, as it does itself in that case.
+    private func provideReceiptEmailIfNecessary(from order: Order) -> String? {
+        let paymentsPluginsDataProvider = PaymentsPluginsDataProvider()
+
+        let wcPay = paymentsPluginsDataProvider.getWCPayPlugin()
+        let stripe = paymentsPluginsDataProvider.getStripePlugin()
+
+        guard !paymentsPluginsDataProvider.bothPluginsInstalledAndActive(wcPay: wcPay, stripe: stripe) else {
+            // This case should not happen, shall we fatal error here?
```

Comment:
My 1ct - yes better to fatal, because this means that something really wrong goes in the code and the feature won't work anyway

Reply:
A silent error is not ideal. I do not love the idea of crashing on purpose, but at the very least, we should log the reason why we stop execution, and we should try to inform the user of what is going on. 

In this particular case, we have seen this case happening in production. There have been users with both WCPay and the Stripe plugin installed, so we should guard against it properly.

Reply:
Thank you all for the feedback 🎉 ! As it seems we have these options:

1. Fatal error and crash: Pros: Quick and meaningful feedback. Cons: Not a good user experience especially in a critical user case (payment).
2. Payment Error:
           - Silently: Pros: Less code and complexity to implement Cons: Not a good experience for the user, who might not know what is happening.
           - With a UI Message: Pros: Good user experience, communicate to the user what is happening. Cons: Payment fails (is it necessary?)
3. Go with the payment but return nil receipt email: Pros: Less code and complexity, the payment goes on. Cons: Would the payment work if both of them are installed and active? If that is the case, I would go for this one for this hotfix and create an issue to properly handle this case on the payment in the future.

Reply:
The outcome of the discussion:
- For the receipt email case, we do not send it if we have WCPay and Stripe both installed and active at the same time. We do not fail or crash.
- We implement the [onboarding checks](https://href.li/?https://github.com/woocommerce/woocommerce-ios/issues/6607) in another PR